From patchwork Mon Apr 15 22:21:04 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Nadav Amit <nadav.amit@gmail.com>
X-Patchwork-Id: 10901651
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id B0246139A
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Apr 2019 22:21:10 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 95DED28517
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Apr 2019 22:21:10 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 868B328893; Mon, 15 Apr 2019 22:21:10 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 3E6ED28517
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Apr 2019 22:21:10 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727437AbfDOWVI (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Mon, 15 Apr 2019 18:21:08 -0400
Received: from mail-pg1-f196.google.com ([209.85.215.196]:33239 "EHLO
        mail-pg1-f196.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726609AbfDOWVI (ORCPT <rfc822;kvm@vger.kernel.org>);
        Mon, 15 Apr 2019 18:21:08 -0400
Received: by mail-pg1-f196.google.com with SMTP id k19so9285007pgh.0
        for <kvm@vger.kernel.org>; Mon, 15 Apr 2019 15:21:08 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=R7o/HxXcrSf1dUo+93q7ZbQq6cPsJsXuEsenie7hfRU=;
        b=Y72Cpd8DzdV1ccpXJv/apk15EhXYs4gSBIl9wZV735MvYpofhBYiDLeIHW0sHMLrRW
         oKJCVundQ8Wyy/2obNMsdrOLvFAKtYf5KcD4ufuYK6f72MKR2Ej7vUxtaYseb1pKeuh6
         ouVdxccS8BWMyiWUGa5yaSTkSKT4R5sVkPIqiwhrHWph+u6ethXvx8S87+K2i9SRk8Vv
         sdmdgXPVDdyzBA9pEyaKnNR/qCOP8XHO9zagNmMIuw3VAp0fMTuAoWcIACXUYVYpPhKa
         yz7QpiGPKqDDva8ybwtjfCoWuG0dzc2aoQh29ns0lXujjeIEo0gGAFRK8wRmMCicdAbb
         FqnQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=R7o/HxXcrSf1dUo+93q7ZbQq6cPsJsXuEsenie7hfRU=;
        b=QXdwZ4W96LFcepHQMTQU/K9IdmL5FQjzk1ycZmDHaf8JNInq4cDNO+9mIuOA4GHE0D
         4pjaPstNz7pvUzLcJve1E/HG/sVkxPUAsOAEkXaTyb5adeogbbI5lEvnm4VB8AMKLYSj
         BUWKQPOgVHNkIL1924Il9ZZrvcN2A/0ah83+2qHeqxVqOuBxUTarokji5tsIQxkjF4ke
         9hpyli+H040em6wH35WGA0FXtGJgjmOHnycdfkXY7MyHugSb7+tP+qeQvE//g7xbAH/s
         06UlUaRD9TzmpECqwG9Sv4LYCTQes8fwll/ylK6KZOhzNUnKp5+4YXlvbMgtFhzQWzqz
         tKig==
X-Gm-Message-State: APjAAAVObREzFIeAIitNiiZKl3xtVGPthNorU9N+7DCRjAmPRedD43Ii
        dU2rnQuHU9OwmTQGacT5V2A=
X-Google-Smtp-Source: 
 APXvYqydQpHgOTJtxQ/z81BbRkOB/OsX2dSs+7MdlfoB6TmTgvGFGERJfxEpI0wRkaIfYHJFXlWl+Q==
X-Received: by 2002:a63:d304:: with SMTP id b4mr72745471pgg.300.1555366866653;
        Mon, 15 Apr 2019 15:21:06 -0700 (PDT)
Received: from htb-2n-eng-dhcp405.eng.vmware.com ([66.170.99.1])
        by smtp.gmail.com with ESMTPSA id
 o5sm149346074pfa.135.2019.04.15.15.21.05
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 15 Apr 2019 15:21:05 -0700 (PDT)
From: nadav.amit@gmail.com
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org, Nadav Amit <nadav.amit@gmail.com>
Subject: [PATCH] x86: Fix allowed IA32_VMX_VMCS_ENUM values
Date: Mon, 15 Apr 2019 15:21:04 -0700
Message-Id: <20190415222104.21367-1-nadav.amit@gmail.com>
X-Mailer: git-send-email 2.19.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Nadav Amit <nadav.amit@gmail.com>

According to the SDM, IA32_VMX_VMCS_ENUM indicates to software the
highest index value used in the encoding of any field supported by the
processor: Bits 9:1 contain the highest index value used for any VMCS
encoding.

Signed-off-by: Nadav Amit <nadav.amit@gmail.com>
---
 x86/vmx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/vmx.c b/x86/vmx.c
index 6ba56bc..f713ada 100644
--- a/x86/vmx.c
+++ b/x86/vmx.c
@@ -1470,7 +1470,7 @@ static void test_vmx_caps(void)
 
 	val = rdmsr(MSR_IA32_VMX_VMCS_ENUM);
 	report("MSR_IA32_VMX_VMCS_ENUM",
-	       (val & 0x3e) >= 0x2a &&
+	       (val & 0x1fe) >= 0x2a &&
 	       (val & 0xfffffffffffffc01Ull) == 0);
 
 	val = rdmsr(MSR_IA32_VMX_EPT_VPID_CAP);
